pkg/gce-pd-csi-driver/node.go

```diff
@@ -866,12 +868,32 @@ func (ns *GCENodeServer) GetVolumeLimits(ctx context.Context) (int64, error) {
 			}
 		}
 	}
+	// Process gen4 A4X machine attach limits, which have a -1g/-2g/-4g/metal suffix
+	if strings.HasPrefix(machineType, "a4x-") {
+		machineTypeSlice := strings.Split(machineType, "-")
+		if len(machineTypeSlice) < 3 {
+			return volumeLimitBig, fmt.Errorf("unconventional machine type: %v", machineType)
+		}
+		gpuString := machineTypeSlice[2]
+		if gpuString == "metal" {
+			return a4xMetalHyperdiskLimit, nil
+		}
+		gpuString = gpuString[0 : len(gpuString)-1] // Remove the 'g' suffix
+		gpus, err := strconv.ParseInt(gpuString, 10, 64)
+		if err != nil {
+			return volumeLimitBig, fmt.Errorf("invalid gpuString %s for machine type: %v", gpuString, machineType)
+		}
+		return common.GetHyperdiskAttachLimit("a4x", gpus), nil
+	}
 	if strings.HasPrefix(machineType, "x4-") {
 		return x4HyperdiskLimit, nil
 	}
 	if strings.HasPrefix(machineType, "a4-") {
 		return a4HyperdiskLimit, nil
 	}
+	if strings.HasPrefix(machineType, "c3-") && strings.HasSuffix(machineType, "-metal") {
```

Comment:
I am not sure if we can add this here? Because c3 VMs still support PD disks, and their attach limit is 128